torch_xla/csrc/runtime/stablehlo_composite_helper.cc

```diff
@@ -190,35 +190,64 @@ class BuildStableHLOCompositePass : public mlir::OperationPass<mlir::ModuleOp> {
     return metadata;
   }
 
+  mlir::FailureOr<mlir::Attribute> BuildAttrFromJson(mlir::OpBuilder& builder,
+                                                     const json& json_value) {
+    switch (json_value.type()) {
+      case json::value_t::number_integer:
+      case json::value_t::number_unsigned:
+        return builder.getI64IntegerAttr(json_value.template get<int64_t>());
+      case json::value_t::number_float:
+        return builder.getF32FloatAttr(json_value.template get<float>());
+      case json::value_t::boolean:
+        return builder.getBoolAttr(json_value.template get<bool>());
+      case json::value_t::string:
+        return builder.getStringAttr(json_value.template get<std::string>());
+      case json::value_t::array: {
+        // Check if all the elements in the json array have the same type.
+        std::optional<json::value_t> common_el_type;
+        for (auto& el : json_value) {
+          auto el_type = el.type();
+          if (el_type == json::value_t::number_unsigned) {
+            el_type = json::value_t::number_integer;
+          }
+          if (common_el_type.has_value() && common_el_type.value() != el_type) {
+            return mlir::failure();
+          }
+          common_el_type = el_type;
+        }
+        if (!common_el_type.has_value()) {
+          return builder.getArrayAttr({});
+        }
+        switch (common_el_type.value()) {
+          case json::value_t::number_integer:
+            return builder.getDenseI64ArrayAttr(
+                json_value.template get<llvm::SmallVector<int64_t>>());
+          case json::value_t::number_float:
+            return builder.getDenseF32ArrayAttr(
+                json_value.template get<llvm::SmallVector<float>>());
+          case json::value_t::boolean:
+            return builder.getDenseBoolArrayAttr(
+                json_value.template get<llvm::SmallVector<bool>>());
+          default:
+            return mlir::failure();
```

Comment:
emitError(builder.getUnknownLocation(), ""invalid JSON to MLIR, invalid array type arrays must be 1-D homogeneous arrays of supported primitive types");